tools/compile_tf_graph.py

```diff
@@ -612,16 +766,32 @@ def cond(i, net_vars, acc_tas, seq_len_info=None)
       seq_len_info = None
     else:
       i, net_vars, acc_tas, seq_len_info = loop_vars
-      seq_len_info = rec_layer.create_state_vars_recursive(["end_flag", "dyn_seq_len"], seq_len_info)
-    i = rec_layer.create_state_var("i", i)
+
     with tf.name_scope("state"):
       # See _SubnetworkRecCell.GetOutput.body.
       # net_vars is (prev_outputs_flat, prev_extra_flat), and prev_outputs_flat corresponds to self._initial_outputs,
       # where the keys are the layer names.
       # prev_extra is always expected to be batch-major.
-      prev_outputs_data = [self.layer_data_templates[k].output for k in sorted(self._initial_outputs.keys())]
-      net_vars = rec_layer.create_state_vars_recursive(
-        name_prefix="state", initial_values=net_vars, data_shape=(prev_outputs_data, None))
+      prev_outputs_data = list()
+      choice_dependent = list()
+      for k in sorted(self._initial_outputs.keys()):
+        choice_dependent.append(False)
+        for source_layer in self.net_dict[k]['from']:
+          if source_layer in self.net_dict.keys() and self.net_dict[source_layer]['class'] == ChoiceStateVarLayer.layer_class:
+            choice_dependent[-1] = True
+        prev_outputs_data.append(self.layer_data_templates[k].output)
+
+      def parse_recursive( nvs ):
+        if isinstance(nvs, (tuple, list)):
+          return [ parse_recursive(v) for v in nvs ]
+        if isinstance(nvs, tf.Tensor):
+          return False
+
+      assert len(net_vars) == 2
+      state_choice_dependent = parse_recursive( net_vars[1] )
```

Comment:
Wrong formatting. (PyCharm should tell you that?)

Reply:
check line before

Reply:
I don't understand. The formatting is wrong.